cmd/node-agent/main.go

```diff
@@ -0,0 +1,278 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/coreos/go-systemd/dbus"
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/api/v1"
+	"k8s.io/kubernetes/pkg/client/cache"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/release_1_3"
+	"k8s.io/kubernetes/pkg/client/restclient"
+	"k8s.io/kubernetes/pkg/controller/framework"
+	"k8s.io/kubernetes/pkg/fields"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/wait"
+	"k8s.io/kubernetes/pkg/watch"
+)
+
+const (
+	myPodNameEnvName        = "MY_POD_NAME"
+	desiredConfigAnnotation = "node-agent.alpha.coreos.com/desired-config"
+	currentConfigAnnotation = "node-agent.alpha.coreos.com/current-config"
+	kubeletVersionKey       = "KUBELET_VERSION"
+	kubeletConfigKey        = "KUBELET_CONFIG"
+	kubeletFlagsKey         = "KUBELET_OPTS"
+	kubeletEnvPath          = "/etc/kubernetes/kubelet.env"
+	kubeletService          = "kubelet.service"
+	configMapFlagsKey       = "flags"
```

Comment:
nit: Be more specific (maybe "kubelet-flags") incase we need to overload this configMap later.